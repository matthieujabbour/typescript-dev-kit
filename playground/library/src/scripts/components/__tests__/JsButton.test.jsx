import React from 'react';
import { act } from 'react-dom/test-utils';
import JsButton from 'scripts/components/JsButton';
import { render, unmountComponentAtNode } from 'react-dom';

let container = document.createElement('div');

describe('react/JsButton', () => {
  beforeEach(() => {
    container = document.createElement('div');
    document.body.appendChild(container);
    jest.clearAllMocks();
  });

  afterEach(() => {
    unmountComponentAtNode(container);
    container.remove();
    container = null;
  });

  test('renders correctly - basic', () => {
    act(() => {
      render(<JsButton label="Test" />, container);
    });
    expect(container).toMatchSnapshot();
  });
});
