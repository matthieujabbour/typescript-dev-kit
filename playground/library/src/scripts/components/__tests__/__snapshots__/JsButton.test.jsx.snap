// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`react/JsButton renders correctly - basic 1`] = `
<div>
  <button
    type="button"
  >
    Test
  </button>
</div>
`;
